doc/papers/joss/paper.md

```diff
@@ -0,0 +1,139 @@
+libCEED provides an interface for matrix-free operator description that enables efficient evaluation on a variety of computational device types (selectable at run time).
+
+# libCEED's API
+
+``libCEED``'s Application Programming Interface (API) provides the local action of the
+linear or nonlinear operator without assembling its sparse representation. Let us
+define the global operator as
+
+\begin{align}\label{eq:decomposition}
+A = P^T \underbrace{G^T B^T D B G}_{\text{libCEED}} P \, ,
+\end{align}
+
+where $P$ is the parallel process decomposition operator (external to libCEED) in
+which the degrees of freedom (DOFs) are scattered to and gathered from the different
+compute devices. The operator denoted by $A_L = G^T B^T D B G$ gives the local action
+on a compute node or process, where $G$ is a local element restriction operation that
+localizes DOFs based on the elements, $B$ defines the action of the basis functions
+(or their gradients) on the nodes, and $D$ is the user-defined pointwise function
+describing the physics of the problem at the quadrature points, also called the
+QFunction. QFunctions, which can either be defined by the user or selected from a
+gallery of available built-in functions in the library, are pointwise functions
+that do not depend on element resolution, topology, or basis degree (selectable
+at run time). This easily allows $hp$-refinement studies (where $h$ commonly denotes the average element size and $p$ the polynomial degree of the basis functions in 1D) and $p$-multigrid solvers. libCEED also supports composition of different operators for multiphysics problems and mixed-element meshes (see Fig. \ref{fig:schematic}).
+
+![A schematic of element restriction and basis applicator operators for
+elements with different topology. This sketch shows the independence of QFunctions
+(in this case representing a Laplacian) element resolution, topology, or basis degree.\label{fig:schematic}](img/QFunctionSketch.pdf)
+
+LibCEED is a C99 library with Fortran77 and Python interfaces. The Python interface was developed using the C Foreign Function Interface (CFFI) for Python. CFFI allows to reuse most of the C declarations and requires only a minimal adaptation of some of them. The C and Python APIs are mapped in a nearly 1:1 correspondence. For instance, a ``CeedVector`` object is exposed as ``libceed.Vector`` in Python, and may reference memory that is also accessed via Python arrays from the NumPy [@NumPy] or Numba [@Numba] packages, for handling host or device memory (when interested in GPU computations with CUDA). Flexible pointer handling in libCEED makes it easy to provide zero-copy host and (GPU) device support for any desired Python array container.
```

Comment:
Do we want to reference our SciPy paper here?

Reply:
I can do it as soon as I have time later. Do you want to add the latest developments on the scaling for multigrid methods, and QFunction context? (I think a part from the possible file organization, the meat in PR #596 will not change much, right?)

Reply:
Done and rebased onto `main`.

Reply:
@jeremylt  do you want to add those two things in here? 
I think we forgot to mention the changes in #596 and #579  in the release notes page

Reply:
Since we don't talk about vectors, except briefly for the Python interface, I don't know that the context object merits much discussion.